classes/core/PKPRoutingProvider.php

```diff
@@ -0,0 +1,181 @@
+<?php
+
+namespace PKP\core;
+
+use PKP\core\PKPContainer;
+use Illuminate\Http\Request;
+use PKP\middlewares\HasUser;
+use PKP\middlewares\HasRoles;
+use Illuminate\Routing\Router;
+use PKP\middlewares\HasContext;
+use Illuminate\Routing\UrlGenerator;
+use PKP\middlewares\AllowCrossOrigin;
+use PKP\middlewares\ValidateCsrfToken;
+use PKP\middlewares\RemoveTrailingSlash;
+use Psr\Http\Message\ServerRequestInterface;
+use Illuminate\Routing\RoutingServiceProvider;
+use PKP\middlewares\DecodeApiTokenWithValidation;
+use PKP\middlewares\SetupContextBasedOnRequestUrl;
+use Illuminate\Foundation\Http\Middleware\TrimStrings;
+use Illuminate\Foundation\Http\Middleware\ValidatePostSize;
+use Illuminate\Contracts\Container\BindingResolutionException;
+use Illuminate\Foundation\Http\Middleware\ConvertEmptyStringsToNull;
+use Illuminate\Contracts\Routing\UrlGenerator as UrlGeneratorContract;
+
+class PKPRoutingProvider extends RoutingServiceProvider
+{
+    protected static $globalMiddlewares = [
+        AllowCrossOrigin::class,
+        SetupContextBasedOnRequestUrl::class,
+        DecodeApiTokenWithValidation::class,
+        ValidateCsrfToken::class,
+        RemoveTrailingSlash::class,
+        ValidatePostSize::class,
+        TrimStrings::class,
+        ConvertEmptyStringsToNull::class,
+    ];
+
+    /**
+     * The application's route middleware.
+     * These middleware can/should be assigned to specific routes or routes groups individually.
+     */
+    protected array $routeMiddlewares = [
+        'has.roles'     => HasRoles::class,
+        'has.user'      => HasUser::class,
+        'has.context'   => HasContext::class,
+    ];
+
+    public static function getGlobalRouteMiddlewares(): array
+    {
+        return self::$globalMiddlewares;
+    }
+
+    /**
+     * Register the service provider.
+     *
+     * @return void
+     */
+    public function register()
```

Comment:
This could present a maintenance headache as it duplicates and extends the parent implementation rather than using `parent::register()`. Would it be possible to let `parent::register()` take care of the built-ins, and have this function just add the missing pieces (`registerRouteMiddlewards()` at a glance)? Or is the order important?

Reply:
> Would it be possible to let parent::register() take care of the built-ins,

Yes 

> Or is the order important?

No 

Thanks for spotting it . It was an unintended left over from routing test locally . Fixed .